File: leetcode/arrays/26-remove-duplicates-from-sorted-array/my_solution.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        index = 0
        prev_num = None
        skipJLoop = False

        for i in range(len(nums)):
            skipJLoop = False
            if prev_num != nums[i]:
                if i != (len(nums)-1):
                    for j in range(i+1, len(nums)):
                        if skipJLoop == False:
                            nums[index] = nums[i]  # assign duplicate to the first of array
                            prev_num = nums[i]     # set the previous number
                            index = index + 1
                            skipJLoop = True
                else:
                    nums[index] = nums[i]
                    prev_num = nums[i]
                    index = index + 1
                    skipJLoop = True
        
        # nums = nums[:index]
        return index


# Runtime: 243 ms, faster than 5.84% of Python3 online submissions for Remove Duplicates from Sorted Array.
# Memory Usage: 15.7 MB, less than 46.00% of Python3 online submissions for Remove Duplicates from Sorted Array.

File: leetcode/arrays/118-pascals-triangle/my-solution.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        // MY BRUTEFORCE SOLUTION
        
        // Array list to store the final pascal triangle row as lists
        List<List<Integer>> pascalsTriangle = new ArrayList();
        // print 'n' lines of pascal's triangle series
        int n=0;
        while(n < numRows) {
            // list to store the row of the pascal's triangle elements
            List<Integer> row = new ArrayList();
            
            // starting of the pascal's triangle
            if(n==0) {
                row.add(1);
            } else {
                // the 'previous-row' of the pascal's triangle
                List<Integer> lastArray = pascalsTriangle.get(pascalsTriangle.size() -1);
                
                // append 1 first to the list(next row)
                row.add(1);
                for(int i=1; i<lastArray.size(); i++) {
                    // append the sum of the previous element and the current element to the list(new row)
                    row.add(lastArray.get(i) + lastArray.get(i-1));
                }
                // append 1 last to the list(next row)
                row.add(1);
            }
            
            // add every row as a separate list to the resultant list
            pascalsTriangle.add(row);
            n++;
        }
        
        return pascalsTriangle;
    }
}

// Runtime: 1 ms, faster than 42.48% of Java online submissions for Pascal's Triangle.
// Memory Usage: 38.2 MB, less than 43.08% of Java online submissions for Pascal's Triangle.